### PR TITLE
Feature/stp kick at pos tactic

### LINKED ISSUE #855: KickAtPos
Assume you have the ball
Rotate around ball to get to correct shooting position
Shoot

Failure: lose ball

### PR DESCRIPTION
**Pull request by timovdk**
Added the KickAtPos condition #855

### CODE REVIEW

File: include/roboteam_ai/stp/new_tactics/KickAtPos.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    /**
+     * Calculate the kickforce for the skill
+     * @param distance distance to the target
+     * @param desiredBallSpeedType type of the kick
+     * @return the speed the kicker needs to kick at
+     */
+    double determineKickForce(double distance, KickChipType desiredBallSpeedType) noexcept;
+
+    virtual bool isEndTactic() noexcept override;
```
Comment (timovdk) [inline include/roboteam_ai/stp/new_tactics/KickAtPos.h]:
```suggestion
    bool isEndTactic() noexcept override;
```

File: src/stp/new_skills/Rotate.cpp [hunk 2]
```diff
@@ -19,7 +19,7 @@ Status Rotate::onUpdate(const StpInfo &info) noexcept {
     int targetDribblerSpeed = targetDribblerPercentage / 100.0 * Constants::MAX_DRIBBLER_CMD();
 
     // Set angle command
-    command.set_w(targetAngle.getAngle());
+    command.set_w(targetAngle);
```
Comment (jordiluong) [inline src/stp/new_skills/Rotate.cpp]:
```suggestion
    command.set_w(targetAngle.getAngle());
```
targetAngle is of Angle class now

File: src/stp/new_skills/Rotate.cpp [hunk 3]
```diff
@@ -28,7 +28,7 @@ Status Rotate::onUpdate(const StpInfo &info) noexcept {
 
     // Check if successful
     double errorMargin = Constants::GOTOPOS_ANGLE_ERROR_MARGIN() * M_PI;
-    if (info.getRobot().value()->getAngle().shortestAngleDiff(targetAngle) < errorMargin) {
+    if (fabs(info.getRobot().value()->getAngle() - targetAngle) < errorMargin) {
```
Comment (jordiluong) [inline src/stp/new_skills/Rotate.cpp]:
```suggestion
    if (info.getRobot().value()->getAngle().shortestAngleDiff(targetAngle) < errorMargin) {
```
Angle class

Comment (timovdk) [reply src/stp/new_skills/Rotate.cpp]: This fix broke the skill

**Review by RobotJente — APPROVED** [review]
Looks good

**Review by Ratoone — DISMISSED** [review]
Similar to chip at pos: consider merging development and removing redundant ball updates

File: src/stp/new_tactics/KickAtPos.cpp [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    double angleToBall = (info.getPosition().second - info.getBall()->get()->getPos()).angle();
+    skillStpInfo.setAngle(angleToBall);
+
+    // Calculate the distance and the kick force
+    double distanceBallToTarget = (info.getBall()->get()->getPos() - info.getPosition().second).length();
+    skillStpInfo.setKickChipVelocity(determineKickForce(distanceBallToTarget, skillStpInfo.getKickChipType()));
+
+    // When rotating, we need to dribble to keep the ball, but when kicking we don't
+    if(skills.current_num() == 0) {
+        skillStpInfo.setDribblerSpeed(31);
```
Comment (timovdk) [inline src/stp/new_tactics/KickAtPos.cpp]:
```suggestion
        skillStpInfo.setDribblerSpeed(100);
```

**Pull request merged**